controllers/tmf-apis/usageManagement.js

```diff
@@ -0,0 +1,138 @@
+var AccountingService = require('./../../db/schemas/accountingService'),
+    async = require('async'),
+    storeClient = require('./../../lib/store').storeClient,
+    utils = require('./../../lib/utils');
+
+var usageManagement = ( function () {
+
+    /**
+     * Check if the API Key passed is a valid API Key
+     *
+     * @param  {Object}   req      Incoming request.
+     */
+    var validateApiKey = function (req, callback) {
+
+        var apiKey = req.get('X-API-KEY');
+
+        if (!apiKey) {
+
+            return callback({
+                status: 401,
+                message: 'Missing header "X-API-KEY"'
+            });
+
+        } else {
+
+            AccountingService.findOne( {apiKey: apiKey}, function (err, result) {
+
+                if (err) {
+
+                    return callback({
+                        status: 500,
+                        message: 'Error validating apiKey'
+                    });
+
+                } else if (!result) {
+
+                    return callback({
+                        status: 401,
+                        message: 'Invalid apikey'
+                    });
+
+                } else if (result.state !== 'COMMITTED') {
+
+                    return callback({
+                        status: 401,
+                        message: 'Apikey uncommitted'
+                    });
+
+                } else {
+                    return callback();
+                }
+            });
+        }
+    };
+
+    /**
+     * Assigns the user id to the relatedPaty query string if it is not defined.
+     *
+     * @param  {Object}   req      Incoming request
+     */
+    var validateRetrieving = function (req, callback) {
+
+        var userId = req.user.id;
+
+        if (!req.query.relatedParty) {
+
+            req.query.relatedParty = {
+                id: userId
+            };
+
+        } else {
+
+            if (req.query.relatedParty.id !== userId) {
+
+                return callback({
+                    status: 401,
+                    message: 'Invalid relatedParty'
+                });
+            }
+        }
+
+        return callback();
+    };
+
+    // If the usage notification to the usage management API is successful, 
+    //  it will notify the the Store with the API response
+    var executePostValidation = function (req, callback) {
+
+        try {
+            var body = JSON.parse(req.body)
```

Comment:
This function is only called when the server returns `200 OK` so there should not be any error parsing the body. If an error arises, an error should be passed in the callback.